p2p/src/main/kotlin/maru/p2p/P2PNetworkBuilder.kt

```diff
@@ -0,0 +1,248 @@
+    val gossipPeerScoreParams = GossipPeerScoreParams()
+    val gossipScoreParams =
+      GossipScoreParams(
+        gossipPeerScoreParams,
+        gossipTopicsScoreParams,
+      )
+    val gossipParams = GossipParamsBuilder().heartbeatInterval(100.millis).build()
+
+    gossipTopicHandlers.add("topic", TestTopicHandler())
+
+    val gossipRouterBuilder = GossipRouterBuilder()
+    gossipRouterBuilder.params = gossipParams
+    gossipRouterBuilder.scoreParams = gossipScoreParams
+    gossipRouterBuilder.messageFactory = { getMessageFactory(it) }
+    val gossipRouter = gossipRouterBuilder.build()
+
+    val pubsubApiImpl = PubsubApiImpl(gossipRouter)
+    val gossip = Gossip(gossipRouter, pubsubApiImpl)
+
+    val metricsSystem = NoOpMetricsSystem() // TODO: add if it is needed
+
+    val publisherApi = gossip.createPublisher(privateKey, Random.nextLong())
+
+    val gossipNetwork =
+      LibP2PGossipNetwork(
+        metricsSystem,
+        gossip,
+        publisherApi,
+        gossipTopicHandlers,
+      )
+
+    val peerId = privateKey?.let { PeerId.fromPubKey(it.publicKey()) }
+    val libP2PNodeId = LibP2PNodeId(peerId)
+
+    val metricTrackingExecutorFactory = MetricTrackingExecutorFactory(metricsSystem)
+    val asyncRunner = AsyncRunnerFactory.createDefault(metricTrackingExecutorFactory).create("maru", 2)
+    val maruRpcMethod = rpcMethod // TODO: MaruRpcMethod needs to be implemented
+    val rpcHandler = RpcHandler(asyncRunner, maruRpcMethod)
+
+    this.peerManager =
+      PeerManager(
+        metricsSystem,
+        ReputationManager.NOOP,
+        mutableListOf<PeerHandler>(MaruPeerHandler()),
+        mutableListOf(rpcHandler),
+        { _ -> 50.0 },
+      ) // TODO need to implement the scoring function
+
+    maruRpcMethod.setPeerManager(peerManager!!)
+
+    val host = createHost(privateKey!!, listOf(gossip, peerManager!!), gossip, rpcHandler)
+    // Add additional connection handlers and protocol bindings as needed
+
+    val advertisedAddresses = mutableListOf<Multiaddr>()
+    ipv4Address?.let { advertisedAddresses.add(it) }
+    ipv6Address?.let { advertisedAddresses.add(it) }
+
+    val libP2PNetwork =
+      LibP2PNetwork(
+        privateKey,
+        libP2PNodeId,
+        host,
+        peerManager,
+        advertisedAddresses,
+        gossipNetwork,
+        mutableListOf(1), // TODO: this does not seem to be used at all
+      )
+
+    return libP2PNetwork
+  }
+
+  private fun getMessageFactory(msg: Rpc.Message): PreparedPubsubMessage {
+    //      com.google.common.base.Preconditions.checkArgument(
+    //        msg.topicIDsCount == 1,
+    //        "Unexpected number of topics for a single message: " + msg.topicIDsCount
+    //      )
+    //      val arrivalTimestamp: Optional<UInt64> = if (recordArrivalTime) {
+    //      Optional.of<UInt64>(timeProvider.getTimeInMillis())
+    //    } else {
+    //      Optional.empty<UInt64>()
+    //    }
```

Comment:
looks like leftover testing code that can be removed

Reply:
gone